cmd/snap/cmd_run.go

```diff
@@ -1509,6 +1536,25 @@ func (x *cmdRun) runSnapConfine(info *snap.Info, runner runnable, beforeExec fun
 				return err
 			}
 		}
+
+		// If a journal namespace is supplied for the service, then we reopen stdout/stderr
+		// connected to that journal namespace instead of the main journal. Since we are not
+		// using systemd's LogNamespace= directly, we must do this ourselves.
+		if lns := os.Getenv("SNAPD_LOG_NAMESPACE"); lns != "" {
+			stdout, stderr := makeStdStreamsForJournal(app, lns)
+			if stdout != nil {
+				defer stdout.Close()
+				if err := sys.Dup2(stdout.Fd(), uintptr(syscall.Stdout)); err != nil {
+					logger.Noticef("cannot duplicate stdout for connection: %v", err)
+				}
+			}
+			if stderr != nil {
+				defer stderr.Close()
+				if err := sys.Dup2(stderr.Fd(), uintptr(syscall.Stderr)); err != nil {
+					logger.Noticef("cannot duplicate stderr for connection: %v", err)
+				}
+			}
```

Comment:
it's prob ok to unset SNAPD_LOG_NAMESPACE here

Reply:
good point